app/src/main/java/sopt/org/millie/presentation/search/searchresult/SearchViewModel.kt

```diff
@@ -1,20 +1,44 @@
 package sopt.org.millie.presentation.search.searchresult
 
+import android.util.Log
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import okhttp3.internal.toImmutableList
+import sopt.org.millie.R
+import sopt.org.millie.core.util.UiState
+import sopt.org.millie.data.repository.SearchRepository
+import sopt.org.millie.presentation.search.navigation.SearchResultRoute
+import sopt.org.millie.presentation.search.searchresult.book.model.SearchBookModel
+import sopt.org.millie.presentation.search.searchresult.library.model.SearchLibraryModel
 import javax.inject.Inject
+import kotlin.Long
+import kotlin.String
 
 @HiltViewModel
 class SearchViewModel
     @Inject
-    constructor() : ViewModel() {
+    constructor(
+        savedStateHandle: SavedStateHandle,
+        private val searchRepository: SearchRepository,
+    ) : ViewModel() {
     private val _uiState = MutableStateFlow(SearchUiState())
     val uiState: StateFlow<SearchUiState> = _uiState.asStateFlow()
 
+    val keyword = savedStateHandle.toRoute<SearchResultRoute>().keyword
+
+    init {
+        loadSearchResult(keyword)
+        loadLibraryList()
+    }
```

Comment:
```suggestion
    init {
        loadSearchResult(keyword)
        updateText(keyword)
        loadLibraryList()
    }
```
이거 추가 안해주면 홈 -> 검색 결과 이동시 검색 결과 화면에 검색 키워드 안뜰텐디 !

Reply:
그 부분은 생각을 못했네용..ㅎㅎ 알려주셔서 감사합니다!